### DISCUSSION TITLE
No more multi_access() - New Secured Tokens and Sync events

### DISCUSSION THREAD

**Discussion by hassandraga** [webui-dev/webui#238] [General]
Because I noticed some weird things... I recently was working locally during nights and weekends in a "_heavy_" core changes (https://github.com/webui-dev/webui/commit/39f9ab0cd75866de37014f644193eded9dd541b6, https://github.com/webui-dev/webui/commit/53724d4ef87dc56fc19b14514824cc925d1a42a6, https://github.com/webui-dev/webui/commit/e462d7bfed87fd2a7e8883cb686d2e6a66791f04).

Old issues:
- A simple URL copy/paste can make you close the main window
- An easily modified version of `webui.js` embedded in a local `test.htm` can make you control the values of the main window
- Calling a back-end function multiple times quickly can send the same first response to all other requests
- Enabling `multi_access()` it just allows showing the UI in the other windows. That's it. But ALL event's response goes to the first window only... that's because WebUI is window-based and NOT client-based design... otherwise, we will be forced to add `client_id` to all functions

So, after spending days fixing all those issues, WebUI now is better than before, stable and secure, and no more weird things.
However, I plan to add `TLS` and `multi-arguments` and do a final test to release v2.4.0.

**Comment by hassandraga** [comment]
About TLS implementation let's continue [here](https://github.com/webui-dev/webui/issues/237).

**Comment by fibodevy** [comment]
Since there is now option to start the web server without opening a browser, `multi_access` is not needed.

**Reply by hassandraga** [reply]
That's true, but I should point out that only one page/tab can access the server using the URL. The other tabs will get blocked, and even a modified `webui.js` will get blocked too by tokens.
